esphome/components/mt6701/mt6701_sensor.cpp

```diff
@@ -0,0 +1,48 @@
+#include "mt6701_sensor.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace mt6701 {
+
+static const char *const TAG = "mt6701";
+static const uint8_t REGISTER_ANGLE = 0x03;
+
+float position2degrees(uint16_t pos) {
+  const uint16_t resolution = 0x4000;  // 14 bits
+  return (pos * 360.0f) / resolution;
+}
+
+void MT6701Sensor::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MT6701 '%s'...", this->name_.c_str());
+
+  if (!read_position().has_value()) {
+    this->mark_failed();
```

Comment:
```suggestion
  if (!this->read_position_().has_value()) {
    this->mark_failed(READ_FAILED_MESSAGE);
```
Log a reason what happened. It will get printed on config dump. Error message is stored.